src/NgramModel-Tests/NgramModelTest.class.st

```diff
@@ -34,6 +34,32 @@ NgramModelTest >> testEmptyModelOfOrder2HasOrder2 [
 	self assert: model order equals: 2.
 ]
 
+{ #category : #tests }
+NgramModelTest >> testEntireTrainedModelCounts [
+	| model tokens |
+	model := NgramModel order: 2.
+	tokens := {'lorem ipsum dolor sit amet' substrings. 'lorem ipsum sit' substrings.
+			'lorem dolor amet' substrings}.
+	model trainOn: tokens.
+	
+	self assert: (model countOfNgram: #(lorem ipsum) asNgram) equals: 2.
```

Comment:
Add an assertion for n-gram that did not appear in the training set: #(ipsum lorem).
And another one that has words that never appeared before: #(hello world).

You can put those into a separate test method(-s)